docker-compose.yml

```diff
@@ -15,5 +15,23 @@ services:
     environment:
       MYSQL_DATABASE: starchart
       MYSQL_USER: starchart
-      MYSQL_PASSWORD: starchart_password 
+      MYSQL_PASSWORD: starchart_password
       MYSQL_ROOT_PASSWORD: root_password
+
+  webhook:
+    # a docker in docker container that installs go, installs webhook, and runs it
+    image: docker:dind
```

Comment:
I think we should write a Dockerfile and build an Alpine image that has:

1. Defines the ENVs for passing in the token
2. [Docker](https://wiki.alpinelinux.org/wiki/Docker).  We'll have to match the version of Docker in the host.
3. [webhook](https://alpine.pkgs.org/3.15/alpine-community-x86_64/webhook-2.8.0-r4.apk.html).  I would pin it to a version vs. use `latest`, so you don't get a surprise when they update and it breaks.
4. copies the deploy script and hooks.json into the image
5. runs the webhook server as its CMD, using port 8000
6. exposes 8000

It's probably worth creating a top-level `webhook/` directory, and you can put your Dockerfile in there, along with the script, JSON, etc.

I also wonder if it's worth doing this in `docker-compose.yml` (i.e., the development compose), or if we should create `production.yml` and do it there.

Reply:
Sure, mounting the `hooks/` directory works.  What if you do:

```
webhook/
  Dockerfile
  hook/
    hooks.json
    *.sh
```

Reply:
Resolved. The new custom image expects a directory to be mounted to `/hooks` with the `hooks.yaml` file in its root.

Using YAML instead of JSON because its designed to be human readable, and we can have comments.